### DISCUSSION TITLE
Improve documentation on how to create maven plugins

### DISCUSSION THREAD

**Discussion by hendrikebbers** [support-and-care/maven-support-and-care#5] [Ideas for Funding]
With the introduction of Maven 4 the documentation of Maven is very outdated. A lot of the internals of Maven have changed. Here the documentation on how Maven Plugins work should be updated. This documentation will help people to contribute to existing Maven plugins and create new plugins.

**Comment by khmarbaise** [comment]
A full description how to setup a development for a plugin
 * Requirements
 * Dependencies
 * How to test
 * How to access dependencies? 
 * How to access directories (based on conventions) (injections)
 * Life Cycle Binding / Command Line ?

**Comment by hboutemy** [comment]
official doc exists https://maven.apache.org/plugin-developers/index.html

yes, it has to be improved, and Maven 4 plugins need to be covered

but please don't create a new "official" documentation that competes with existing one the https://xkcd.com/927/ way

**Reply by khmarbaise** [reply]
Of course the idea to improve or reorder the documentation and NOT to create another **official** doc, because some aspects of writing a plugin is not clear.. For example how to handle cases  like:
* I would like to get the dependencies 
* I would like to get information about the modules / reactor or alike
* What is an aggregate goal? (good idea?)
* How to write an reporting plugin?
* (TBD more ideas)

**Comment by ascheman** [comment]
This discussion has been converted to Epic #57 for tracking.

Closing this discussion.
